atlp-rwanda/e-comm-team-techsmith-fn
# #185218911 Ch-setup ESlint pre commit

## What does this PR do?
- Setup ESLint 
- Configure prettier
- Integrate prettier with ESLint
- Implement Git pre commit hook

## Description of Task to be completed?
#### Configure ESLint and Git Pre commit hook to run tests and ESLint format before commits

## How should this be manually tested?

- Clone the repository using `git clone` [link](https://github.com/atlp-rwanda/e-comm-team-techsmith-fn)

- Checkout to the branch using `git checkout <branch-name>`

- Install node packages that the application depends on: `npm i`
- To test ESLint and pre-commit create a file inside src folder and write codes which violates some ESLint rules 
- In your terminal run git command `git add .` and then try `git commit -m "your commit message"` you will see how we run lint script to check errors and format script to format our code if any error is found the commit process will be aborted

## Any background context you want to provide?

## What are the relevant pivotal tracker/Trello stories?
- #185218911

## Screenshots (if appropriate)
